src/libraries/ANALYSIS/DEventWriterROOT.cc

```diff
@@ -714,6 +714,7 @@ void DEventWriterROOT::Create_Branches_ChargedHypotheses(DTreeBranchRegister& lo
 	locBranchRegister.Register_FundamentalArray<Float_t>(Build_BranchName(locParticleBranchName, "SigTrans_BCAL"), locArraySizeString, dInitNumNeutralArraySize);
 	locBranchRegister.Register_FundamentalArray<Float_t>(Build_BranchName(locParticleBranchName, "RMSTime_BCAL"), locArraySizeString, dInitNumNeutralArraySize);
 	locBranchRegister.Register_FundamentalArray<Float_t>(Build_BranchName(locParticleBranchName, "Energy_FCAL"), locArraySizeString, dInitNumNeutralArraySize);
+	locBranchRegister.Register_FundamentalArray<Float_t>(Build_BranchName(locParticleBranchName, "Energy_FCAL_SingleHit"), locArraySizeString, dInitNumNeutralArraySize);
```

Comment:
I don't quite see the point of duplicating these field at the ROOT Tree level (e.g. two energy entries).  If so, I think there should be some command-line flag to control if this is being written out?  But maybe it would be better to just write out the number of hits that go into the shower, and that should be enough to distinguish the one-block from the normal showers?  This info is saved in the REST files

Reply:
Good point -- this is probably worth fixing.

Reply:
Also for DOCA to track

Reply:
the current code looks good to me